pycbc/distributions/constraints.py

```diff
@@ -37,7 +38,12 @@ def __init__(self, constraint_arg, static_args=None, transforms=None,
                 static_args.items(), key=lambda x: len(x[0]), reverse=True))
             )
         for arg, val in static_args.items():
-            constraint_arg = constraint_arg.replace(arg, str(val))
+            swp = f"'{val}'" if isinstance(val, str) else str(val)
+            # Substitute static arg name for value if it appears in the
+            # constraint_arg string at the beginning of a word and is not
+            # followed by an underscore or equals sign
+            constraint_arg = re.sub(
+                r'\b{}(?!\_|\=)'.format(arg), swp, constraint_arg)
```

Comment:
@SamuelH-97, can you remind us about why this change was introduced?

@cdcapano, do you see this potentially breaking `constraints.py` applications?

Reply:
I'm afraid this change was made by @a-r-williamson and we never went over the reason for it so I'm not quite certain

Reply:
@titodalcanton and I looked through some old tests of Andrew's and we understood that this regex call handles things like this:

"""
[static_params]
eos = 2H

[constraint-3]
name = custom
constraint_arg = remnant_mass_from_mass1_mass2_spherical_spin_eos(mass1, mass2, spin1_a, spin1_polar, eos=eos, solve=True) > 0
"""

This ensures that each injection in the injection file has a record of what eos was used (2H) and that the call to `remnant_mass_from_mass1_mass2_spherical_spin_eos` uses the keyword `eos` with value `2H` rather than `eos`.

Reply:
Sounds good.